Source/Scene/Scene.js

```diff
@@ -2829,44 +2905,51 @@ define([
         }
 
         frameState.creditDisplay.endFrame();
-
-        if (scene.debugShowFramesPerSecond) {
-            if (!defined(scene._performanceDisplay)) {
-                var performanceContainer = document.createElement('div');
-                performanceContainer.className = 'cesium-performanceDisplay-defaultContainer';
-                var container = scene._canvas.parentNode;
-                container.appendChild(performanceContainer);
-                var performanceDisplay = new PerformanceDisplay({container: performanceContainer});
-                scene._performanceDisplay = performanceDisplay;
-                scene._performanceContainer = performanceContainer;
-            }
-
-            scene._performanceDisplay.update();
-        } else if (defined(scene._performanceDisplay)) {
-            scene._performanceDisplay = scene._performanceDisplay && scene._performanceDisplay.destroy();
-            scene._performanceContainer.parentNode.removeChild(scene._performanceContainer);
-        }
-
         context.endFrame();
-        RequestScheduler.update();
-        callAfterRenderFunctions(frameState);
-
-        scene._postRender.raiseEvent(scene, time);
     }
 
     /**
      * @private
      */
-    Scene.prototype.render = function(time) {
+    Scene.prototype.render = function (time) {
+        if (!defined(time)) {
+            time = JulianDate.now();
+        }
+
+        this._preRender.raiseEvent(this, time);
+        this._jobScheduler.resetBudgets();
+
+        var cameraChanged = checkForCameraUpdates(this);
+
+        var shouldRender = !this.requestRenderMode || this._renderRequested || cameraChanged || (this.mode === SceneMode.MORPHING);
+
+        var now = JulianDate.clone(time);
+        if (!shouldRender && defined(this.maximumRenderTimeChange) && defined(this._lastRenderTime)) {
+            var difference = Math.abs(JulianDate.secondsDifference(this._lastRenderTime, now));
+            shouldRender = shouldRender || difference >= this.maximumRenderTimeChange;
+        }
+
         try {
-            render(this, time);
+            if (shouldRender) {
+                this._lastRenderTime = now;
+                this._renderRequested = false;
+
+                render(this, time);
+            } else {
+                update(this, time);
+            }
         } catch (error) {
             this._renderError.raiseEvent(this, error);
 
             if (this.rethrowRenderErrors) {
                 throw error;
             }
         }
+
+        updateDebugShowFramesPerSecond(this, shouldRender);
+        RequestScheduler.update();
+        callAfterRenderFunctions(this._frameState);
+        this._postRender.raiseEvent(this, time);
```

Comment:
Should the after render functions and events only happen if a render has actually occurred? They should probably stay here, but I'm not sure.

Reply:
If the listeners subscribed to these events do not take into account time in the callback function, this could cause some unexpected issues. If they do take into account time, it will advance correctly even when render was not called. To be safe, I would prefer to leave them here, but naming is not ideal. We could rename them from "render" to "update"?

Reply:
Ok, It might be worth updating the doc for `scene.postRender` to indicate that the event is raised every `update` instead of each `render`. I wonder if that qualifies as a breaking change...?

Reply:
It won't break any functionality without explicitly enabling `requestRenderMode`. Should I just note that in the documentation then? I'm not sure either if this qualifies as a breaking change. @pjcozzi ?

Reply:
Please review the use cases for this event and decide the right course of action - seems that it currently couples render and perhaps time change and/or update - which may call for separate events and reasonable deprecation.  No need for anymore input from me.